fix: email patch failure (#15106)

mtraeber opened the pull request:
This should solve the issue for the patch (https://github.com/frappe/frappe/issues/15106).

As I understood @ankush  right, then the patch needs an `frappe.reload_doc()` first to update the schema

Added `frappe.reload_doc()` of both DocTypes `Email Account` and `Email Flag Queue`
Correct the field name in `Email Flaq Queue`-update and added it into `Email Flaq Queue`-DocType.


caused by: https://github.com/frappe/frappe/pull/12608

mtraeber committed:
Merge branch 'frappe:develop' into develop

mtraeber committed:
Merge branch 'frappe:develop' into develop

mtraeber committed:
Merge branch 'frappe:develop' into develop

mtraeber committed:
Merge branch 'frappe:develop' into develop

mtraeber committed:
Fix: copy_email_data patch #15106

Reload DocType `Email Account` in patch.
Changed `modified` in json files.

Removed linking of `imap_folder` in `Email Flag Queue` this connection
seems not necessary at this point. Also removed all parts that create
this connection.

ankush commented:
`reload_doc` won't work as the `modified` key hasn't been updated in, I'll add a fix on this PR.

ankush commented on frappe/patches/v14_0/copy_mail_data.py:
```diff
@@ -21,6 +23,6 @@ def execute():
 		EmailFlagQueue = frappe.qb.DocType("Email Flag Queue")
 		frappe.qb.update(EmailFlagQueue) \
-			.set(EmailFlagQueue.imap_folder, "INBOX") \
+			.set(EmailFlagQueue.folder_name, "INBOX") \
```
This patch fixes existing queue, what about new records?

https://github.com/frappe/frappe/blob/72fa9a3c1980cbaf29d307928382a5c466a4aa51/frappe/email/inbox.py#L87-L94 

I am unsure of this. @leela can you review this part once?

surajshetty3416 replied:
The code has been removed... I don't see `folder_name` in `Email Flag Queue` doctype... 
It seems that part of the code is irrelevant now.

Also, `folder_name` should not be necessary to update the "read" status of an email... **UID** should be enough.

ankush commented:
@mtraeber `reload_doc` won't work as two doctypes don't have updated `modified` key

Please update `modified` to today's date in `email_account.json` and `communication.json`, that will fix the patch failure. I tried to push but github won't let me, maybe you've disabled maintainer edits on this PR 😅

mtraeber committed:
Merge branch 'frappe:develop' into fix_email_patch

mtraeber commented:
@ankush mhm, i have nothing changed on github settings, but now i think it's activated.

Changed `modified` in json files, also i thought about the `Email Flag Queue` connection and i think this is not necessary and removed all parts that want to save the `folder_name` in `Email Flag Queue`. 

But let's wait for a review of this part.

thanks!

ankush commented:
@mtraeber patch fix looks okay. 

Can someone review/confirm Flag Queue related changes? @surajshetty3416 @gavindsouza   https://github.com/frappe/frappe/pull/15115#discussion_r758938427

surajshetty3416 commented:
https://github.com/frappe/frappe/pull/15115#discussion_r760142538

LGTM. Merging.
